extensions/query-store/src/settings/configureDialog.ts

```diff
@@ -0,0 +1,626 @@
+	private criteriaRadioButtons?: azdata.RadioButtonComponent[]
+	private consumptionCriteriaComponent?: azdata.FormComponent<azdata.Component>;
+	private formBuilder?: azdata.FormBuilder;
+	private formModel?: azdata.FormContainer;
+	private criteriaBasisAvgRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisMaxRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisMinRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisStdDevRadioButton?: azdata.RadioButtonComponent;
+	private criteriaBasisTotalRadioButton?: azdata.RadioButtonComponent;
+	private executionCountCheckBox?: azdata.CheckBoxComponent;
+	private durationCheckBox?: azdata.CheckBoxComponent;
+	private CPUTimeCheckBox?: azdata.CheckBoxComponent;
+	private logicalReadsCheckBox?: azdata.CheckBoxComponent;
+	private logicalWritesCheckBox?: azdata.CheckBoxComponent;
+	private physicalReadsCheckBox?: azdata.CheckBoxComponent;
+	private CLRTimeCheckBox?: azdata.CheckBoxComponent;
+	private DOPCheckBox?: azdata.CheckBoxComponent;
+	private memoryConsumptionCheckBox?: azdata.CheckBoxComponent;
+	private rowCountCheckBox?: azdata.CheckBoxComponent;
+	private showChartComponent?: azdata.FormComponent<azdata.Component>;
+	private timeIntervalOptionsDropdown?: azdata.DropDownComponent;
+	private customTimeFromTextBox?: azdata.InputBoxComponent;
+	private customTimeToTextBox?: azdata.InputBoxComponent;
+	private localTimeFormatRadioButton?: azdata.RadioButtonComponent;
+	private UTCTimeFormatRadioButton?: azdata.RadioButtonComponent;
+	private timeIntervalComponent?: azdata.FormComponent<azdata.Component>;
+	private aggregationSizeComponent?: azdata.FlexContainer;
+	private timeIntervalOptionsOverallReport?: string[];
+	private timeIntervalOptionsTopResourceReport?: string[];
+	private returnDataAllRadioButton?: azdata.RadioButtonComponent;
+	private returnDataTopRadioButton?: azdata.RadioButtonComponent;
+	private returnDataTopInputBox?: azdata.InputBoxComponent;
+	private returnComponent?: azdata.FormComponent<azdata.Component>;
+	private filtersInputBox?: azdata.InputBoxComponent;
+	private filterComponent?: azdata.FormComponent<azdata.Component>;
+	private initDialogComplete?: Deferred<void>;
+	private initDialogPromise: Promise<void> = new Promise<void>((resolve, reject) => this.initDialogComplete = { resolve, reject });
+
+	constructor() {
+		this.dialog = azdata.window.createModelViewDialog(constants.configure);
+		this.configureTab = azdata.window.createTab(constants.configure);
+		this.dialog.registerCloseValidator(async () => {
+			return this.validate();
+		});
+	}
+
+	async validate(): Promise<boolean> {
+		return true;
+	}
+
+	public async openDialog(): Promise<void> {
+		await this.initializeDialog();
+
+		this.dialog.okButton.label = constants.okButtonText;
+		this.dialog.okButton.enabled = false;
+		this.toDispose.push(this.dialog.okButton.onClick(async () => { }));		// TODO: Add return of settings change and functionality to enable OK Button
```

Comment:
the ok, cancel, and apply onClicks should call dispose()

Reply:
As far as I can understand, apply shouldn't be calling dispose- because the dialog is still open, so shouldn't yet be disposed.
As for, OK button click- adding dispose call (and the call exists for Cancel button).